### ISSUE TITLE
Use newer gibson (i.e. to enable sticky sessions)

### ISSUE THREAD

**Issue by MatthiasWinzeler** [cloudfoundry-incubator/route-registrar#6]
It would be great if you could bump the dependency to a newer version (current master?) of [https://github.com/cloudfoundry/gibson](gibson). 

This would enable the use of sticky sessions: https://github.com/cloudfoundry/gibson/pull/5

I wanted to create a PR, but gave up because the tests on the current `route-registrar` master always fail; see also the broken [https://travis-ci.org/cloudfoundry-incubator/route-registrar/builds/92790401](Travis build #42). Besides this error (caused by not using godep I assume), I get different errors on each run, i.e.:

```
• Failure [0.026 seconds]
Registrar.RegisterRoutes [It] Sends a router.register message and does not send a router.unregister message
/home/vagrant/gopath/src/github.com/cloudfoundry-incubator/route-registrar/registrar/registrar_test.go:85

  Failed after 0.010s.
  Expected
      <chan bool | len:0, cap:0>: 0xc208032540
  not to receive anything.
```

or 

```
Registrar.RegisterRoutes When the registrar has a healthchecker [It] Emits a router.unregister message when registrar's health check fails, and emits a router.register message when registrar's health check back to normal
/Users/taawima0/git/gopath/src/github.com/cloudfoundry-incubator/route-registrar/registrar/registrar_test.go:129

  Timed out after 3.001s.
  Expected
      <chan string | len:0, cap:0>: 0xc208034960
  to receive something
```

**Comment by robdimsdale** [comment]
Hi @MatthiasWinzeler 

We have refactored the code and tests so they are no longer intermittently failing with non-deterministic results. We have also made some backward-incompatible changes to the config.

These changes are not yet on the master branch, as we are waiting for a PR to gibson to be merged. See the [WIP branch](https://github.com/cloudfoundry-incubator/route-registrar/tree/WIP-110684012) for more details.

The version of gibson in Godeps will be updated when we merge this WIP branch - this should address this issue.

It is likely that we will move away from travis in the future; we will update the README accordingly when we do so.

We will keep this issue open until our changes are on master; feel free to update this issue if you feel it won't be addressed by our changes.

Thanks,
Rob && @zaksoup

**Comment by robdimsdale** [comment]
@MatthiasWinzeler

We removed the dependency on gibson and yagnats. Instead we now use github.com/nats-io/nats

As a result we will close this issue.

Thanks

Rob && @zaksoup

**Comment by Amit-PivotalLabs** [comment]
@MatthiasWinzeler 

We are adding features to this codebase to deregister routes, and condition maintaining the route registration contingent on a health check.  See:

* https://www.pivotaltracker.com/story/show/107466938
* https://www.pivotaltracker.com/story/show/107467058

In service of these features, we had to switch away from gibson and yagnats to nats-io/nats.  If you have separate reasons for wanting to bump (e.g. sticky sessions), could you open up a separate issue please?

Thanks,
Amit

**Comment by MatthiasWinzeler** [comment]
Thanks for the heads-up. Sticky sessions already work with master (since (you're setting privateInstanceId when announcing the route)[https://github.com/cloudfoundry-incubator/route-registrar/blob/master/registrar/registrar.go#L54]), so it works fine for me.
